modules/gatk4/variantfiltration/main.nf

```diff
@@ -35,7 +35,7 @@ process GATK4_VARIANTFILTRATION {
     gatk VariantFiltration \\
         -R $fasta \\
         -V $vcf \\
-        -O ${prefix}.vcf \\
+        -O ${prefix}.vcf.gz \\
         $options.args
```

Comment:
is there any way to specify number of cpus etc.?

Reply:
There is a way for the spark version of GATK. May be we can see this later?

Reply:
ok that then requires  a different module I would say. One other thing, for the GATK we have these lines generally:

```
def avail_mem       = 3
    if (!task.memory) {
        log.info '[GATK HaplotypeCaller] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
    } else {
        avail_mem = task.memory.giga
    }
```

Some gatk thing I believe to make sure enough memory is available.

Reply:
Ok yes, I can add that as well.

Reply:
@FriederikeHanssen if we use `avail_mem` inside Xmx as below, the test fails. I think it breaks the quotes
`gatk --java-options "-Xmx${avail_mem}g" \\`

I have seen some modules that define the variable, but NOT used anywhere. why is it so? One example [here](https://github.com/nf-core/modules/blob/master/modules/gatk4/markduplicates/main.nf)

Reply:
You are right,  it needs to be specified somehow. I will add it to n existing issue on gatk memory. 
Can you try it without quotes like [here](https://github.com/nf-core/atacseq/blob/1b3a832db5a53c92c2c89a4d6d79455e860461ad/main.nf#L739)

Reply:
Just removing quotes did not work. But it is solved with the followed way:

`gatk --java-options "-Xmx${avail_mem}G"`

Recommendation here: https://gatk.broadinstitute.org/hc/en-us/articles/360035531892-GATK4-command-line-syntax
Resubmitted PR

Reply:
great 👍